src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,505 @@
+			return jsonParamValueInjection;
+		}
+
+		public String[][] getUrlEncTimedValueInjection() {
+			return urlEncTimedValueInjection;
+		}
+		
+		public String getIndexDb() {
+			return indexDb;
+		}
+	};
+
+   private static final Logger LOG = Logger.getLogger(TestNoSQLInjection.class);
+
+	@Override
+	public int getId() {
+		return ID;
+	}
+
+	 @Override
+	 public boolean targets(TechSet technologies) {
+		if (technologies.includes(Tech.Db)) {
+			return true;
+		}
+
+		for (Tech tech : technologies.getIncludeTech()) {
+			if (tech.getParent() == Tech.Db) {
+				return true;
+			}
+		}
+		return false;	
+	 }
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name." + indexDbSel);
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc." + indexDbSel);
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln." + indexDbSel);
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs." + indexDbSel);
+	}
+	
+	@Override
+	public String[] getDependency() {
+		return new String[] {};
+	}
+   
+	@Override
+	public int getCweId() {
+		return CWED_ID;
+	}
+
+	@Override
+	public int getWascId() {
+		return WASC_ID;
+	}
+	
+	public String getExtraInfo(String attack) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack + "." + indexDbSel);
+	}
+	
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+
+		try {			
+			for(NOSQLDB nosql: NOSQLDB.values()) {
+				
+				indexDbSel = nosql.getIndexDb();
+				
+				if(inScope(nosql.getDbTech())){
+					startUrlEncodedScan(nosql, msg, param, value);
+					startJsonScan(nosql, msg, param, value);
+				}
+			}
+		} catch (SocketException ex) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+						" when accessing: " + msg.getRequestHeader().getURI().toString());
+			}
+		} catch (IOException ex) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+						" when accessing: " + msg.getRequestHeader().getURI().toString());
+			}
+		} catch(JSONException ex) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+						" when try to convert the payload in the json format");
+			}
+		}
+	}
+
+	private void startUrlEncodedScan(NOSQLDB nosql, HttpMessage msg, String param, String value) throws IOException {
+		
+		HttpMessage baseMsg;
+		long intervalBaseMessage, intervalInjectedMessage; 
+		Instant now;
+		baseMsg = getNewMsg();
+		setParameter(baseMsg, param, TOKEN);
+		now = Instant.now();
+		sendAndReceive(baseMsg);
+		intervalBaseMessage = ChronoUnit.MILLIS.between(now, Instant.now());

```

Comment:
The previously established `now` variable should probably be renamed `start` to avoid confusion and increase readability.